pysph/examples/solid_mech/oscillating_plate.py

```diff
@@ -0,0 +1,170 @@
+
+
+class OscillatingPlate(Application):
+    def initialize(self):
+        self.L = 0.2
+        self.H = 0.02
+        # wave number K
+        self.KL = 1.875
+        self.K = 1.875 / self.L
+
+        # edge velocity of the plate (m/s)
+        self.Vf = 0.05
+        self.dx_plate = 0.002
+        self.h = 1.3 * self.dx_plate
+        self.plate_rho0 = 1000.
+        self.plate_E = 2. * 1e6
+        self.plate_nu = 0.3975
+
+        # bulk modulus
+        self.plate_inside_wall_length = self.L / 4.
+        self.wall_layers = 3
+
+        self.alpha = 1.0
+        self.beta = 1.0
+        self.xsph_eps = 0.5
+        self.artificial_stress_eps = 0.3
+
+        self.tf = 1.0
+        self.dt = 1e-5
+
+    def create_particles(self):
+        # plate coordinates
+        xp, yp = np.mgrid[-self.plate_inside_wall_length:self.L +
+                          self.dx_plate / 2.:self.dx_plate, -self.H /
+                          2.:self.H / 2. + self.dx_plate / 2.:self.dx_plate]
+        xp = xp.ravel()
+        yp = yp.ravel()
+
+        m = self.plate_rho0 * self.dx_plate**2.
+
+        # get the index of the particle which will be used to compute the
+        # amplitude
+        xp_max = max(xp)
+        fltr = np.argwhere(xp == xp_max)
+        fltr_idx = int(len(fltr) / 2.)
+        amplitude_idx = fltr[fltr_idx][0]
+
+        kernel = CubicSpline(dim=2)
+        self.wdeltap = kernel.kernel(rij=self.dx_plate, h=self.h)
+        plate = get_particle_array_elastic_dynamics(
+            x=xp, y=yp, m=m, h=self.h,
+            rho=self.plate_rho0, name="plate", constants=dict(
+                wdeltap=self.wdeltap, n=4, rho_ref=self.plate_rho0,
+                E=self.plate_E, nu=self.plate_nu, amplitude_idx=amplitude_idx))
+
+        ##################################
+        # vertical velocity of the plate #
+        ##################################
+        # initialize with zero at the beginning
+        v = np.zeros_like(xp)
+        v = v.ravel()
+
+        # set the vertical velocity for particles which are only
+        # out of the wall
+        K = self.K
+        # L = self.L
+        KL = self.KL
+        M = sin(KL) + sinh(KL)
+        N = cos(KL) + cosh(KL)
+        Q = 2 * (cos(KL) * sinh(KL) - sin(KL) * cosh(KL))
+        for i in range(len(v)):
+            if xp[i] > 0.:
+                # set the velocity
+                tmp1 = (cos(K * xp[i]) - cosh(K * xp[i]))
+                tmp2 = (sin(K * xp[i]) - sinh(K * xp[i]))
+                v[i] = self.Vf * plate.cs[0] * (M * tmp1 - N * tmp2) / Q
+
+        # set vertical velocity
+        plate.v = v
+
+        # #########################################
+        # #### Create the wall particle array #####
+        # #########################################
+        # get the minimum and maximum of the plate
+        xp_min = xp.min()
+        yp_min = yp.min()
+        yp_max = yp.max()
+        xw_upper, yw_upper = np.mgrid[-self.plate_inside_wall_length:self.
```

Comment:
This block is hard to read, use some temporary variables like `dx` etc. to make this easier to read.